hikey.mk

```diff
@@ -1,12 +1,35 @@
+################################################################################
+# User-defined variables
+# Edit so these match your target
+# NOTE: If making changes after a build, please clean before rebuilding!
+################################################################################
+# Non-secure user mode (root fs binaries): 32 or 64-bit [default 64]
 NSU ?= 64
-SU ?= 32
+# Secure kernel (OP-TEE OS): 32 or 64-bit [default 64]
 SK ?= 64
+# Secure user mode (Trusted Apps): 32 or 64-bit [default 32, requires SK=64 for 64]
+SU ?= 32
+
+# Console UARTs: 3 or 0
```

Comment:
I don't really like the idea of setting all this individually. I mean either you set it to be UART0 or UART3? In the end I would like to, from make, call:
```
make
# Building with both NW and SW at UART0 (default)
```
And ...
```
make CFG_HIKEY_UART3=y
# Building with NW UART3 and SW at UART0
```
This approach will hardcode the secure side to UART0, but do we think we will use something else? I.e, is there any need for making that configurable using a make argument?

So, what about using a global FLAG something like this ... psuedo code
```
CFG_HIKEY_UART ?= 0
ARM_TF_CONSOLE_UART ?= $CFG_HIKEY_UART
OPTEE_CONSOLE_UART ?= 0
EDK2_CONSOLE_UART ?= $CFG_HIKEY_UART
LINUX_CONSOLE_UART ?= $CFG_HIKEY_UART
```

Also I probably would like to move the different ARM_..., OPTEE_..., EDK2_... and LINUX_... down to the section about the component itself.